packages/lib-classifier/src/plugins/drawingTools/experimental/components/LineControls/LineControls.js

```diff
@@ -33,11 +32,22 @@ const LineControls = forwardRef(function LineControls({
   const OUTER_RADIUS = 40 / scale
   const INNER_RADIUS = 20 / scale
 
-  const { viewBox, rotate, width, height } = useContext(SVGContext)
-  const [x, y, w, h] = viewBox.split(' ').map(n => parseInt(n, 10))
+  const { canvas } = useContext(SVGContext)
+
+  // Get the transformation that's been applied to a parent element
+  const matrix = canvas.parentElement.parentElement.getAttribute('transform');
+  const regex = /\((.*?)\)/;
+  const match = matrix.match(regex);
+  const [hScale, hSkew, vSkew, vScale, hTranlate, vTranslate] = match[1].split(', ').map(s => parseInt(s, 10))
```

Comment:
You might get better performance here if you get the matrix directly from the SVG DOM.

```suggestion
  // Get the transformation that's been applied to a parent element
  const { matrix } = canvas.parentElement.parentElement.transform.baseVal.consolidate()
  const { a: hScale, b: hSkew, c: vSkew, d: vScale, e: hTranslate, f: vTranslate } = matrix
```